mcs/class/corlib/Test/System/StringTest.cs

```diff
@@ -92,19 +96,36 @@ public void Constructor4_Count_Negative ()
 		}
 	}
 
+	// Several tests in this file, to run properly, allocate 4GB objects.
+	// Obviously this creates problems on several kinds of systems, so we
+	// conservatively skip these tests unless we find a high-RAM environment.
+	// Checking RAM requires PerformanceCounter which is absent on mobile,
+	// so any test that calls this must be category MobileNotWorking.
+	static bool LowMemoryTestEnvironment ()
+	{
+#if MOBILE
+		return true;
+#else
+		if (!Environment.Is64BitProcess)
+			return true;
+
+		// Require 6 GB physical RAM, for the 4GB string plus 2GB headroom 
+		var pc = new PerformanceCounter ("Mono Memory", "Total Physical Memory");
+
+		return pc.RawValue < 6L*1024L*1024L*1024L;
```

Comment:
I'm gonna leave this as if if that's ok, the Ls are harmless and for me they save the 0.25s of valuable time trying to remember the int promotion rules when scanning the code.